contracts/contracts/strategies/NativeStaking/ValidatorRegistrator.sol

```diff
@@ -0,0 +1,345 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { Pausable } from "@openzeppelin/contracts/security/Pausable.sol";
+import { Governable } from "../../governance/Governable.sol";
+import { IDepositContract } from "../../interfaces/IDepositContract.sol";
+import { IVault } from "../../interfaces/IVault.sol";
+import { IWETH9 } from "../../interfaces/IWETH9.sol";
+import { ISSVNetwork, Cluster } from "../../interfaces/ISSVNetwork.sol";
+
+struct ValidatorStakeData {
+    bytes pubkey;
+    bytes signature;
+    bytes32 depositDataRoot;
+}
+
+/**
+ * @title Registrator of the validators
+ * @notice This contract implements all the required functionality to register, exit and remove validators.
+ * @author Origin Protocol Inc
+ */
+abstract contract ValidatorRegistrator is Governable, Pausable {
+    /// @notice The maximum amount of ETH that can be staked by a validator
+    /// @dev this can change in the future with EIP-7251, Increase the MAX_EFFECTIVE_BALANCE
+    uint256 public constant FULL_STAKE = 32 ether;
+
+    /// @notice The address of the Wrapped ETH (WETH) token contract
+    address public immutable WETH;
+    /// @notice The address of the beacon chain deposit contract
+    address public immutable BEACON_CHAIN_DEPOSIT_CONTRACT;
+    /// @notice The address of the SSV Network contract used to interface with
+    address public immutable SSV_NETWORK;
+    /// @notice Address of the OETH Vault proxy contract
+    address public immutable VAULT_ADDRESS;
+    /// @notice Maximum number of validators that can be registered in this strategy
+    uint256 public immutable MAX_VALIDATORS;
+
+    /// @notice Address of the registrator - allowed to register, exit and remove validators
+    address public validatorRegistrator;
+    /// @notice The number of validators that have 32 (!) ETH actively deposited. When a new deposit
+    /// to a validator happens this number increases, when a validator exit is detected this number
+    /// decreases.
+    uint256 public activeDepositedValidators;
+    /// @notice State of the validators keccak256(pubKey) => state
+    mapping(bytes32 => VALIDATOR_STATE) public validatorsStates;
+    /// @notice The account that is allowed to modify stakeETHThreshold and reset stakeETHTally
+    address public stakingMonitor;
+    /// @notice Amount of ETH that can be staked before staking on the contract is suspended
+    /// and the governor needs to approve further staking
+    uint256 public stakeETHThreshold;
+    /// @notice Amount of ETH that can has been staked since the last governor approval.
+    uint256 public stakeETHTally;
+    // For future use
+    uint256[47] private __gap;
```

Comment:
Similar gap comment - we have six storage slot variables used at this layer, so our gap should be 44 slots?

Reply:
same as the other one. It started at 50 when we first deployed to Holesky but with changes it's now 47